doc/specs/#1595 - Suggestions UI/Snippets.md

```diff
@@ -0,0 +1,622 @@
+
+The initial version of the snippets pane:
+
+![snippets-pane](https://github.com/microsoft/terminal/assets/18356694/f4aee6f8-dbaa-4a70-a2ac-98244eb0e4f1)
+
+## Tenets
+
+<table>
+
+<tr><td><strong>Compatibility</strong></td><td>
+
+I considered supporting YAML for local snippets (`.wt.json`), instead of JSON.
+JSON is not super friendly to command-lines - since everything's gotta be
+encapsulated as a string. Embedding tabs `\t`, newlines `\r`, escape characters,
+is fairly straightforward. However, quotes can get complicated fast in JSON,
+since they've got to be escaped too, and with many CLI utilities also having
+separate quote-parsing rules, JSON can get unwieldy quickly.
+
+However, supporting YAML directly would require us to spec out a YAML syntax for
+these files, and also find an OSS YAML parser and implement support for it. That
+would be quite a bit more expensive than JSON.
+
+</td></tr>
+
+<tr><td><strong>Accessibility</strong></td><td>
+
+Nothing particular to call out here. The Snippets pane will need to be a11y
+tested, like most of our other UI surfaces.
+
+</td></tr>
+
+<tr><td><strong>Sustainability</strong></td><td>
+
+No substantial climate impacts expected here. We're not using expensive compute
+resources for this feature, so the impact should be comparable to any other
+Terminal feature.
+
+</td></tr>
+
+<tr><td><strong>Localization</strong></td><td>
+
+I'm mildly worried here about the potential for community-driven tasks to have
+non-localized descriptions. We may need to accept a `description:{ en-us:"",
+pt-br:"", ...}`-style map of language->string descriptions. That may just need
+to be a future consideration for now.
+
+</td></tr>
+
+<tr><td><a name="security"></a><strong>Security</strong></td><td>
+
+Another reason we shouldn't support keys being able to be lazy-bound to local
+snippets: It's entirely too easy for `malicious.exe` to create a file in
+`%HomePath%` that creates a snippet for `\u003pwn-your-machine.exe\r` (or
+similar). Any app can read your settings file, and it is again too easy for that
+malicious app to set it's own action `id` to the same as some other well-meaning
+local snippet's ID which you DO have bound to a key.
+
+When we first load the snippets from the `.wt.json` file, we'll want to also ask
+them if they trust that folder. This is similar to the way that VsCode et. al.
+If they accept, then we'll add that folder to a list of trusted folders (and
+store permanently in `state.json`). If they don't, then we'll just ignore that
+file. To make things easier for the user, we can also add a checkbox to "trust
+the parent folder" in the dialog (again, similar to VsCode).
+
+We'll also want to engage our security partners to see if there's anything extra
+we'll need to do to ensure we're securely parsing random JSON files that we
+don't own.
+
+</td></tr>
+
+</table>
+
+### Other potential issues

```

Comment:
How does this work with `ssh`? Or is that something we don't have to worry about?